### PR TITLE
fix(run-protocol): fix add liquidity check

### LINKED ISSUE #5503: Add liquidity fails
## Describe the bug
When trying to add liquidity on sim-chain `agoric start --reset` on a pool that does in fact exist, the offer fails with:

```
2022-06-03T05:08:08.268Z fake-chain: delivering to sim-chain (trips=110)
2022-06-03T05:08:08.858Z SwingSet: ls: v20: Logging sent error stack (TypeError#4)
2022-06-03T05:08:08.859Z SwingSet: ls: v20: TypeError#4: addLiquidity: no function
2022-06-03T05:08:08.859Z SwingSet: ls: v20: TypeError: addLiquidity: no function
 at addLiquidity (.../run-protocol/src/vpool-xyk-amm/addLiquidity.js:30)
 at apply ()
```

### PR DESCRIPTION
**Pull request by samsiegart**
fixes: https://github.com/Agoric/agoric-sdk/issues/5503

### CODE REVIEW

**Comment by turadg** [comment]
Note this would have been caught by https://github.com/Agoric/agoric-sdk/pull/5495 :)

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/run-protocol/src/vpool-xyk-amm/addLiquidity.js [hunk 1]
```diff
@@ -25,9 +26,9 @@ const makeMakeAddLiquidityInvitation = (zcf, getPool) => {
 
     // Get the brand of the secondary token so we can identify the liquidity pool.
     const secondaryBrand = seat.getProposal().give.Secondary.brand;
-
     const pool = getPool(secondaryBrand);
-    const liquidityBrand = pool.getLiquidityIssuer().getBrand();
+    const liquidityIssuer = E(pool).getLiquidityIssuer();
+    const liquidityBrand = await E(liquidityIssuer).getBrand();
```
Comment (dtribble) [inline packages/run-protocol/src/vpool-xyk-amm/addLiquidity.js]: This is not a good reason for making the offer hook async and have an `await`.  There should be an API that provides the brand from the ZCF, which should be synchronous.

**Review by dtribble — CHANGES_REQUESTED** [review]

File: packages/run-protocol/src/vpool-xyk-amm/addLiquidity.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/run-protocol/src/vpool-xyk-amm/addLiquidity.js]: If you have the issuer inside a contract, you can directly call `zcf.getBrandForIssuer(issuer)` to get the brand synchronously.

Comment (samsiegart) [reply packages/run-protocol/src/vpool-xyk-amm/addLiquidity.js]: Just got that working, thanks. PTAL.

**Review by Chris-Hibbert — APPROVED** [review]

**Review by turadg — APPROVED** [review]

**Review by dtribble — APPROVED** [review]
LGTM

**Comment by samsiegart** [comment]
Thanks for reviews!

**Pull request merged**